src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java

```diff
@@ -42,4 +43,41 @@ public LeaseAcquirer(LeaseClientProvider leaseClientProvider) {
             return Optional.empty();
         }
     }
+
+    public void ifAcquiredOrElse(BlobClient blobClient, Runnable onSuccess, Runnable onFailure) {
+        try {
+            var leaseClient = leaseClientProvider.get(blobClient);
+            leaseClient.acquireLease(LEASE_DURATION_IN_SECONDS);
+
+            onSuccess.run();
+
+            release(leaseClient, blobClient);
+
+        } catch (BlobStorageException exc) {
+            if (exc.getErrorCode() != BlobErrorCode.LEASE_ALREADY_PRESENT) {
```

Comment:
If lease is already present, would it be worth logging it as info? Just wondering if that information would be useful, e.g. to see that we tried to progress with the file.

Reply:
this is copied from existing method `acquireFor`

Reply:
Yeah, not strictly related to this PR.